pkg/cmd/openshift-operators/webconsole-operator/webconsole_operator_trigger.go

```diff
@@ -0,0 +1,275 @@
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/util/workqueue"
+	apiregistrationclientv1beta1 "k8s.io/kube-aggregator/pkg/client/clientset_generated/clientset/typed/apiregistration/v1beta1"
+
+	operatorsv1alpha1 "github.com/openshift/origin/pkg/cmd/openshift-operators/apis/operators/v1alpha1"
+	"github.com/openshift/origin/pkg/cmd/openshift-operators/apis/operators/v1alpha1helpers"
+	webconsoleclientv1alpha1 "github.com/openshift/origin/pkg/cmd/openshift-operators/generated/clientset/versioned/typed/webconsole/v1alpha1"
+	webconsoleinformerv1alpha1 "github.com/openshift/origin/pkg/cmd/openshift-operators/generated/informers/externalversions/webconsole/v1alpha1"
+)
+
+const targetNamespaceName = "openshift-web-console"
+
+type WebConsoleOperator struct {
+	operatorConfigClient webconsoleclientv1alpha1.OpenShiftWebConsoleConfigsGetter
+
+	appsv1Client      appsclientv1.AppsV1Interface
+	corev1Client      coreclientv1.CoreV1Interface
+	apiServicesClient apiregistrationclientv1beta1.APIServicesGetter
+
+	// queue only ever has one item, but it has nice error handling backoff/retry semantics
+	queue workqueue.RateLimitingInterface
+}
+
+func NewWebConsoleOperator(
+	webconsoleConfigInformer webconsoleinformerv1alpha1.OpenShiftWebConsoleConfigInformer,
+	webconsoleNamespacedKubeInformers informers.SharedInformerFactory,
+	operatorConfigClient webconsoleclientv1alpha1.OpenShiftWebConsoleConfigsGetter,
+	appsv1Client appsclientv1.AppsV1Interface,
+	corev1Client coreclientv1.CoreV1Interface,
+) *WebConsoleOperator {
+	c := &WebConsoleOperator{
+		operatorConfigClient: operatorConfigClient,
+		appsv1Client:         appsv1Client,
+		corev1Client:         corev1Client,
+
+		queue: workqueue.NewNamedRateLimitingQueue(workqueue.DefaultControllerRateLimiter(), "WebConsoleOperator"),
+	}
+
+	webconsoleConfigInformer.Informer().AddEventHandler(c.eventHandler())
+	webconsoleNamespacedKubeInformers.Core().V1().ConfigMaps().Informer().AddEventHandler(c.eventHandler())
+	webconsoleNamespacedKubeInformers.Core().V1().ServiceAccounts().Informer().AddEventHandler(c.eventHandler())
+	webconsoleNamespacedKubeInformers.Core().V1().Services().Informer().AddEventHandler(c.eventHandler())
+	webconsoleNamespacedKubeInformers.Apps().V1().Deployments().Informer().AddEventHandler(c.eventHandler())
+
+	// we only watch some namespaces
+	webconsoleNamespacedKubeInformers.Core().V1().Namespaces().Informer().AddEventHandler(c.namespaceEventHandler())
+
+	return c
+}
+
+func (c WebConsoleOperator) sync() error {
+	operatorConfig, err := c.operatorConfigClient.OpenShiftWebConsoleConfigs().Get("instance", metav1.GetOptions{})
+	if err != nil {
+		return err
+	}
+	switch operatorConfig.Spec.ManagementState {
+	case operatorsv1alpha1.Unmanaged:
+		return nil
+
+	case operatorsv1alpha1.Removed:
+		// TODO probably need to watch until the NS is really gone
+		if err := c.corev1Client.Namespaces().Delete(targetNamespaceName, nil); err != nil && !apierrors.IsNotFound(err) {
```

Comment:
> does the operator have special role or just use the privileged client?

Once it becomes orchestrated (next operator on top), then it will have exactly the role it needs.  Until then, it is a cluster-admin in `oc cluster up` clusters.